src/goto-programs/goto_program.h

```diff
@@ -1169,6 +1169,30 @@ struct pointee_address_equalt
   }
 };
 
+template <typename GotoFunctionT, typename Predicate, typename HandlerT>
+void for_each_goto_location_if(
+  GotoFunctionT &&goto_function,
+  Predicate predicate,
+  HandlerT handle)
+{
+  auto &&instructions = goto_function.body.instructions;
+  for(auto it = instructions.begin(); it != instructions.end(); ++it)
+  {
+    if(predicate(it))
+    {
+      handle(it);
+    }
+  }
+}
+
+template <typename GotoFunctionT, typename HandlerT>
+void for_each_goto_location(GotoFunctionT &&goto_function, HandlerT handle)
```

Comment:
It's called `for_each_goto_location` but inside you iterate over `instructions`.

Reply:
I'm not against these but please note the amount of hatred that `forall_goto_program_instructions` et al. used to get and try to get some kind of agreement on whether we should be doing more or less of this.

Reply:
I've added this here for now, but I don't feel too strongly about this - I'll remove this if people don't like it.

Reply:
I'm fine with it but I know it was a big source of argument at one point (I don't remember what was decided TBH) so be a little careful.  I think it might be in the coding standard...